### DISCUSSION TITLE
Setting repeatable notifications on a specific day?

### DISCUSSION THREAD

**Discussion by srowe0091** [ionic-team/capacitor#2752] [Q&A]
Hello, this is more of a question, than bug as I'm not sure if I have implemented it wrong or need to supply more details in the notification schedule. I am trying to setup a local notification that starts triggering on a specific day, and repeats every 2 weeks, and an added touch will be to set a specific time, so the notification does not go off at 12:00am on the given day.

my environment
```
  @capacitor/ios not installed
  @capacitor/electron not installed
  @capacitor/core 2.0.1
  @capacitor/android 2.0.1
  @capacitor/cli 2.0.1
```

I had put this to start
```
LocalNotifications.schedule({
    notifications: [
      {
        title: "Title",
        body: "Body",
        id: 1,
        schedule: {
          repeats: true,
          every: 'two-weeks',
          on: {
            day: 14
          }
        },
        actionTypeId: "",
        extra: null
      }
    ]
  })
```

and it worked when i manually set the phone to skip 2 weeks in the future, a couple seconds later the notification appeared. But when I tried adding the `hour` property in the `on` it never triggered

```
LocalNotifications.schedule({
    notifications: [
      {
        title: "Title",
        body: "Body",
        id: 1,
        schedule: {
          repeats: true,
          every: 'two-weeks',
          on: {
            day: 14,
            hour: 10
          }
        },
        actionTypeId: "",
        extra: null
      }
    ]
  })
```

So i want the notification to start on the day that i set, for example if today is the 12th, and the date applied is the 15th, i want it to repeat every 2 weeks from the 15th. and setting the time for hour 10AM. Am i missing something with how i tried adding the `hour` prop?

sidenote: is the hour property in 12hr format or 24hr format?

Edit: some more findings, since today is `day 12`, i had set the notification with
```
on: {
   day: 15
}
```
and i set the phones date to the 15th, no trigger. i set it for 2 weeks after 15 (day 29) and it still did not trigger. Suspiciously I tried setting the phones date to 2 weeks after the 12th (day 24), when the notification was created, and the notification WAS triggered. Also oddly enough the time of the phone was set to the morning on `day 24` and it didn't trigger, as it only triggered after 4pm, because that was when i initially created the notification. it seems like the creation of the notification schedule is ignoring the `on` prop entirely

**Comment by dotrub** [comment]
I have been struggling with a similar issue lately. The case I have is even simpler; I want a daily repeating reminder at a time set by the user. I should be able to use `on`, and I do for iOS, but can't on Android because of [this bug](https://github.com/ionic-team/capacitor/issues/2747).

I can answer some of your questions, though. Unfortunately, the best place by far to figure out how LocalNotifications are working is in the source code; the documentation is often inaccurate and provides relatively little info. In the source code, you can see that the schedule options are processed like this:

```
if (at) {
  if (repeats === true) {
    // calculate time between 'now' and 'at'
    // return repeating 'at'-style reminder, starting at 'at' and repeating at the interval calculated
  } else {
    // return 'at'-style reminder (one occurrence)
  }
} else if (every) {
  // return 'every'-style reminder
} else if (on) {
  // return 'on'-style reminder
}
```

So if you provide `at`, it will ignore anything you provide for `every` or `on`. In your case, since you provide `every`, it will ignore `on`. The options do not work in conjunction with one another. And `repeats` only applies to the `at` case, and is applied in a way that frankly makes no sense to me; I can't think of any realistic scenario where that repeating behaviour would be useful.

I tried to implement something like this by scheduling one reminder with `at` and some data in `extra`, intending to use the `localNotificationReceived` listener to then schedule a repeating reminder with `every` from the moment the notification triggered. But upon searching the source code, it appears that there is no implementation for `localNotificationReceived`, at least on Android! Why is it in the docs?! Why isn't it implemented?!

So I'm stumped. I can't think of a reasonable and robust way to accomplish what seems like the simplest base case usage of repeating reminders.

**Reply by elnezah** [reply]
Nice, thanks. Just to share my opinion, I cannot also think on a good scenario for `every` since there is no starting point.  So you'll have a notification every week at the time the user activated it? What is this good for?

**Reply by danielmalmros** [reply]
@dotrub Thanks for the clarification! I have been struggling with this issue for days now. But nothing seems to work for me. I can see this issue is a year old, but still it's hard to figure out how to schedule simple reminders from the docs. Did you manage to come up with a solution? In our use case we need to remind the user every day based on time set by user.  

I have tried to use only `on` but it dose not repeat but instead only appear once.

```
await LocalNotifications.schedule({
  notifications: [
    {
      title: "Title",
      body: "Body text",
      id: uniqueId,
      schedule: {
        on: { hour, minute },
      },
    },
  ],
});
```

**Comment by slevy85** [comment]
I have the same issue, I think a way that could be implemented is like this :

```
schedule: {
          on: {
            dayofweek: 1, // from 0 to 6, 0 is sunday
            hour: 10,
            minute: 30
          }
        },
```
This would trigger the notification every monday at 10:30, but `dayofweek` is not supported.

**Comment by divinetouch** [comment]
I'm having the same issue on Android. I've tried daily, weekly, monthly, and yearly, none of them is working for me.

In my case, I also specify the hour and minute in the schedule property.

**Comment by littke** [comment]
Agree with @slevy85, the `dayofweek` is what's missing here. Then `on` would work for this use case. 

Should be simple enough to implement, as [DateComponents](https://developer.apple.com/documentation/foundation/datecomponents) for iOS already supports `weekday`.

**Comment by danielrf121** [comment]
Good Morning. From what I understand from this discussion, I can't use repetition by setting a specific date and time.

Today I need to repeat reminders, for example, every day at 7 am, to take a medication and this is not possible.

I believe that the repeat setting only makes sense if we set a date and time.

But anyway, reading this thread clarified a lot for me, in these 4 days that I am racking my brains with the use of this plugin.

**Comment by weblaunch** [comment]
So I've spent an afternoon trying to implement a reminder that goes off at X o'clock every weekday. Seems it just can't be done and really I'd expect it to be possible :-/

**Comment by daveshirman** [comment]
I gave up and just did it this way:

https://www.codingandclimbing.co.uk/blog/android-localnotifications-with-cordova-capacitor-example

**Reply by rtpHarry** [reply]
FYI, I know this is an old post but I was just doing research on this for an upcoming project. I had this saved as a possible solution, but it seems that these scheduling issues have been fixed now in https://github.com/ionic-team/capacitor-plugins/pull/756

**Comment by kim-jos** [comment]
I was having the same issue and decided to have a look into the Java files for Android and was able to come up with a solution. 

For those who are still looking for a solution to this problem, you can now implement the weekday feature using the `on` keyword (0 - 6) (Sun - Sat) once you copy and paste it into your project if you need it. https://github.com/ionic-team/capacitor-plugins/pull/601

I made a pull request but I am still waiting for it to get merged.
